src/command/FileCommands.cxx

```diff
@@ -234,9 +236,93 @@ handle_read_comments(Client &client, Request args, Response &r)
 	gcc_unreachable();
 }
 
+AllocatedPath
+find_art(AllocatedPath directory)
+{
+    std::vector<std::string> art_names = {
+        "cover.png",
+        "cover.jpg",
+        "cover.tiff",
+        "cover.bmp"
+    };
+    
+    for(std::string& name: art_names) {
+        const AllocatedPath art = AllocatedPath::Build(directory, name.c_str());
+        
+        FileInfo fi;
+        if (!GetFileInfo(art, fi, false)) {
+            continue;
+        }
+        
+        if (!fi.IsRegular()) {
+            continue;
+        }
+        return art;
+    }
+    return AllocatedPath::Null();
+}
+
 CommandResult
 handle_album_art(Client &client, Request args, Response &r)
 {
-    r.Error(ACK_ERROR_SYSTEM, "Unimplemented");
-    return CommandResult::ERROR;
+    assert(args.size == 1 || args.size == 2);
+    
+    const char *const uri = args.front();
+    unsigned long long offset = 0;
+    if (args.size == 2) {
+        offset = args.ParseUnsigned(1);
+    }
+    
+#ifdef ENABLE_DATABASE
+    const Storage *storage = client.GetStorage();
+#endif
+    const auto located_uri = LocateUri(uri, &client
+#ifdef ENABLE_DATABASE
+                                       , storage
+#endif
+                                       );
+    
+    const AllocatedPath path = storage->MapFS(located_uri.canonical_uri);
+    const AllocatedPath directory = path.GetDirectoryName();
+    
+    AllocatedPath art_name = find_art(directory);
+    if (art_name.IsNull()) {
+        if (offset > 0) {
+            r.Error(ACK_ERROR_NO_EXIST, "No art found");
+            return CommandResult::ERROR;
+        }
+        r.Write("size: -1\n");
```

Comment:
Ok. I guess it's semantics, I'm guess what I'm really saying is that the question the client is asking on first request is "does the file exist?" rather than "just give me the file, I don't care if it exists or not." But if you want to treat it as an error I'm happy with that.